Model/RuleConfigInterface.php

```diff
@@ -0,0 +1,13 @@
+<?php
+declare(strict_types=1);
+
+namespace RedChamps\CleanMenu\Model;
+
+interface RuleConfigInterface
```

Comment:
Shouldn’t be the interface placed in the 'Api' directory?

Reply:
'Api' directory is for service contract declaration and allows third party code to use these services.
In our case RuleConfigInterface if for internal usage within the module. However it could be declared as SPI as third party modules could implement or extend the rules.
So I guess that RuleConfigInterface  could be marked as SPI/, and IsAllowedInterface as API.
What do you think?

Reply:
Make sense. Let's proceed with this. Also, what about ListInterface? should it be SPI as well?